timemachine/cpp/src/context.cu

```diff
@@ -135,6 +139,83 @@ std::array<std::vector<double>, 2> Context::multiple_steps_local(
     return std::array<std::vector<double>, 2>({h_x_buffer, h_box_buffer});
 }
 
+std::array<std::vector<double>, 2> Context::multiple_steps_local_selection(
+    const int n_steps,
+    const int reference_idx,
+    const std::vector<int> &selection_mask,
+    const int burn_in,
+    const int store_x_interval,
+    const double radius,
+    const double k) {
+    if (store_x_interval <= 0) {
+        throw std::runtime_error("store_x_interval <= 0");
+    }
+
+    const int x_buffer_size = n_steps / store_x_interval;
+
+    const int box_buffer_size = x_buffer_size * 3 * 3;
```

Comment:
Isn't the box constant during local MD? (i.e. NVT)

Reply:
Yes, this should be a constant trajectory as currently implemented.

----------------

(Something we haven't discussed, but I think would be technically permissible if motivated, is to interleave MC barostat moves and local MD moves. Right now the context only allows to interleave MC barostat moves with global MD moves, but I do not think this is a real requirement.)

(Also, I don't currently foresee a motivation to apply MC barostat moves _during_ a single local MD move, but that could also be feasible, with some not-yet-worked-out bookkeeping to account for `p(selection | x_scaled, box_scaled) / p(selection| x, box)`. (Haven't investigated yet, but guess would be: incorporate this factor into MC barostat accept/reject step.))

Reply:
It is, though seemed more consistent to return boxes rather than having to have the user handle tracking the boxes that go along with local md.

Reply:
okay - I'm in favor of keeping things consistent then. (but maybe a comment that for NVT boxes are the same)

Reply:
Added comment: https://github.com/proteneer/timemachine/pull/1005/commits/770649816066f30a07ccf1bba6a1dee45f907337